records_mover/db/mysql/load_options.py

```diff
@@ -0,0 +1,200 @@
+                       fail_if_cant_handle_hint: bool) -> MySqlLoadOptions:
+    hints = records_format.hints
+
+    #
+    # The server uses the character set indicated by the
+    # character_set_database system variable to interpret the
+    # information in the file. SET NAMES and the setting of
+    # character_set_client do not affect interpretation of input. If
+    # the contents of the input file use a character set that differs
+    # from the default, it is usually preferable to specify the
+    # character set of the file by using the CHARACTER SET clause. A
+    # character set of binary specifies “no conversion.”
+    #
+    hint_encoding: HintEncoding = hints['encoding']  # type: ignore
+    character_set = MYSQL_CHARACTER_SETS_FOR_LOAD.get(hint_encoding)
+    if character_set is not None:
+        mysql_character_set = character_set
+    else:
+        cant_handle_hint(fail_if_cant_handle_hint, 'encoding', hints)
+        mysql_character_set = 'utf8'
+    quiet_remove(unhandled_hints, 'encoding')
+
+    field_terminator: HintFieldDelimiter = hints['field-delimiter']  # type: ignore
+    mysql_field_terminator = field_terminator
+    quiet_remove(unhandled_hints, 'field-delimiter')
+
+    # https://dev.mysql.com/doc/refman/8.0/en/load-data.html
+    #
+    #
+    # LOAD DATA can be used to read files obtained from external
+    # sources. For example, many programs can export data in
+    # comma-separated values (CSV) format, such that lines have fields
+    # separated by commas and enclosed within double quotation marks,
+    # with an initial line of column names. If the lines in such a
+    # file are terminated by carriage return/newline pairs, the
+    # statement shown here illustrates the field- and line-handling
+    # options you would use to load the file:
+    #
+    # LOAD DATA INFILE 'data.txt' INTO TABLE tbl_name
+    #  FIELDS TERMINATED BY ',' ENCLOSED BY '"'
+    #  LINES TERMINATED BY '\r\n'
+    #  IGNORE 1 LINES;
+    #
+    #
+    mysql_field_enclosed_by = None
+    mysql_field_optionally_enclosed_by = None
+    hint_quotechar: HintQuoteChar = hints['quotechar']  # type: ignore
+    hint_quoting: HintQuoting = hints['quoting']  # type: ignore
+    if hint_quoting == 'all':
+        mysql_field_enclosed_by = hint_quotechar
+    elif hint_quoting == 'minimal':
+        # "If the input values are not necessarily enclosed within
+        # quotation marks, use OPTIONALLY before the ENCLOSED BY option."
+        #
+        # This implies to me that parsing here is permissive -
+        # otherwise unambiguous strings without double quotes around
+        # them will be understood as a string, not rejected.
+        mysql_field_optionally_enclosed_by = hint_quotechar
+    elif hint_quoting == 'nonnumeric':
+        mysql_field_optionally_enclosed_by = hint_quotechar
+    elif hint_quoting is None:
+        pass
+    else:
+        # TODO: Have this be more like _assert_never(hint_quoting)?
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker